contracts/DCNTCrescendo.sol

```diff
@@ -129,8 +133,17 @@ contract DCNTCrescendo is
     return ((bps - takeRateBPS) * _currentPrice[id]) / bps;
   }
 
-  function buy(uint256 id) external payable {
-    require(saleIsActive, "Sale must be active to buy");
+  modifier salesAreActive {
+    require(block.timestamp >= saleStart, "Sales are not active yet.");
+    require(! saleIsPaused, "Sale must be active to buy or sell");
+    _;
+  }
+
+  function saleIsActive() external view returns(bool _saleIsActive) {
+    _saleIsActive = (block.timestamp >= saleStart) && (!saleIsPaused);
+  }
+
+  function buy(uint256 id) external payable salesAreActive {
```

Comment:
```suggestion
  /// @notice purchase nft
  function buy(uint256 id) external payable salesAreActive {
```

could make it simpler on the FE if we call this function `mint` so we don't have to have different UI components / function calls for Crescendo & Editions.